parsers/IN_WE.py

```diff
@@ -108,6 +110,8 @@ def format_exchanges_data(
     filtered_data = filter_raw_data(
         kind="exchange", data=data, target_datetime=target_datetime
     )
+    if filtered_data.empty:
```

Comment:
We do not want to return 0 here and rather a `ParserException`. Returning a 0 here would propagate a "fake" value throughout our entire pipelines. We would rather have a missing datapoint that can later be estimated than a wrong value

Reply:
It wouldn't even propagate as the new ParserClasses will reject them (showing their benefit already!) but we should avoid it anyhow as it is now producing errors instead.

Reply:
fair point but i would rather not create synthetic 0s even at the parser level. It is not good practice :)

Reply:
I cannot throw ParserException here because the exception will break the loop from where this method is called.
I did two changes to handle this properly. 

1. I have updated the get_date_range to generate time range from midnight to **hour of target_datetime**. It will avoid generating dates in the future
2. I am aware of edge cases when the source API has data missing for few hours for some reason, for those cases  the datapoints will not be added to the output

Reply:
If there really are data for future dates that should be taken a look at cause then it might mean this data is actually forecasted and then all data should be returned instead (using the consumptionForecast function) or if they are split between measured and forecasted we should do the same.

Reply:
From this datasource I can confirm there is no future data available, however I agree, that instead of our code deciding which all hours to generate data for from the datasource; our code must simply rely on the data and generate datapoint for hours for which data is available.

In the future, PRs, I will make changes to use **group by on hour** to aggregate data by hour, it will simply generate datapoints for hours for which any data, and skip hours for which data is not there